# Исследование надёжности заёмщиков

Кредитному отделу банка необходимо определить, влияет ли семейное положение и количество детей клиента на погашение кредита в срок. От банка получена статистика о платёжеспособности клиентов.

Результаты исследования будут применены при построении модели кредитного скоринга.

## Шаг 1. Откроем файл с данными и изучим общую информацию

In [1]:
# Загрузим библиотеку Pandas и прочитаем файл.
import pandas as pd
data = pd.read_csv('/datasets/data.csv')
# Посмотрим начало и конец таблицы, общую информацию о таблице и ее размер.
display(data.head(10))
display(data.tail(10))
display(data.info())
display(data.shape)
# Посмотрим, какие значения и в каком количестве есть в отдельных столбцах таблицы.
for column in data.columns:
    display(data[column].value_counts())

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0    -926.185831         27    высшее             0   
6         0   -2879.202052         43    высшее             0   
7         0    -152.779569         50   СРЕДНЕЕ             1   
8         2   -6929.865299         35    ВЫСШЕЕ             0   
9         0   -2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   
5  гражданский брак                 1      M   компаньон     0  255763.565419   
6   женат / замужем                 0      F   компаньон     0  240525.971920   
7   женат / замужем                 0      M   сотрудник     0  135823.934197   
8  гражданский брак                 1      F   сотрудник     0   95856.832424   
9   женат / замужем                 0      M   сотрудник     0  144425.938277   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу  
5               покупка жилья  
6           операции с жильем  
7                 образование  
8       на проведение свадьбы  
9     покупка жилья для семьи

children  days_employed  dob_years       education  education_id  \
21515         1    -467.685130         28         среднее             1   
21516         0    -914.391429         42          высшее             0   
21517         0    -404.679034         42          высшее             0   
21518         0  373995.710838         59         СРЕДНЕЕ             1   
21519         1   -2351.431934         37  ученая степень             4   
21520         1   -4529.316663         43         среднее             1   
21521         0  343937.404131         67         среднее             1   
21522         1   -2113.346888         38         среднее             1   
21523         3   -3112.481705         38         среднее             1   
21524         2   -1984.507589         40         среднее             1   

          family_status  family_status_id gender income_type  debt  \
21515   женат / замужем                 0      F   сотрудник     1   
21516   женат / замужем                 0      F   компаньон     0   
21517  гражданский брак                 1      F   компаньон     0   
21518   женат / замужем                 0      F   пенсионер     0   
21519         в разводе                 3      M   сотрудник     0   
21520  гражданский брак                 1      F   компаньон     0   
21521   женат / замужем                 0      F   пенсионер     0   
21522  гражданский брак                 1      M   сотрудник     1   
21523   женат / замужем                 0      M   сотрудник     1   
21524   женат / замужем                 0      F   сотрудник     0   

        total_income                            purpose  
21515  109486.327999              заняться образованием  
21516  322807.776603               покупка своего жилья  
21517  178059.553491       на покупку своего автомобиля  
21518  153864.650328               сделка с автомобилем  
21519  115949.039788  покупка коммерческой недвижимости  
21520  224791.862382                  операции с жильем  
21521  155999.806512               сделка с автомобилем  
21522   89672.561153                       недвижимость  
21523  244093.050500       на покупку своего автомобиля  
21524   82047.418899              на покупку автомобиля

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


None

(21525, 12)

 0     14149
 1      4818
 2      2055
 3       330
 20       76
-1        47
 4        41
 5         9
Name: children, dtype: int64

-986.927316     1
-7026.359174    1
-4236.274243    1
-6620.396473    1
-1238.560080    1
               ..
-2849.351119    1
-5619.328204    1
-448.829898     1
-1687.038672    1
-582.538413     1
Name: days_employed, Length: 19351, dtype: int64

35    617
40    609
41    607
34    603
38    598
42    597
33    581
39    573
31    560
36    555
44    547
29    545
30    540
48    538
37    537
50    514
43    513
32    510
49    508
28    503
45    497
27    493
56    487
52    484
47    480
54    479
46    475
58    461
57    460
53    459
51    448
59    444
55    443
26    408
60    377
25    357
61    355
62    352
63    269
64    265
24    264
23    254
65    194
66    183
22    183
67    167
21    111
0     101
68     99
69     85
70     65
71     58
20     51
72     33
19     14
73      8
74      6
75      1
Name: dob_years, dtype: int64

среднее                13750
высшее                  4718
СРЕДНЕЕ                  772
Среднее                  711
неоконченное высшее      668
ВЫСШЕЕ                   274
Высшее                   268
начальное                250
Неоконченное высшее       47
НЕОКОНЧЕННОЕ ВЫСШЕЕ       29
НАЧАЛЬНОЕ                 17
Начальное                 15
ученая степень             4
Ученая степень             1
УЧЕНАЯ СТЕПЕНЬ             1
Name: education, dtype: int64

1    15233
0     5260
2      744
3      282
4        6
Name: education_id, dtype: int64

женат / замужем          12380
гражданский брак          4177
Не женат / не замужем     2813
в разводе                 1195
вдовец / вдова             960
Name: family_status, dtype: int64

0    12380
1     4177
4     2813
3     1195
2      960
Name: family_status_id, dtype: int64

F      14236
M       7288
XNA        1
Name: gender, dtype: int64

сотрудник          11119
компаньон           5085
пенсионер           3856
госслужащий         1459
предприниматель        2
безработный            2
студент                1
в декрете              1
Name: income_type, dtype: int64

0    19784
1     1741
Name: debt, dtype: int64

112874.418757    1
133912.272223    1
182036.676828    1
122421.963500    1
198271.837248    1
                ..
133299.194693    1
115080.782380    1
84896.781597     1
153838.839212    1
150014.128510    1
Name: total_income, Length: 19351, dtype: int64

свадьба                                   797
на проведение свадьбы                     777
сыграть свадьбу                           774
операции с недвижимостью                  676
покупка коммерческой недвижимости         664
операции с жильем                         653
покупка жилья для сдачи                   653
операции с коммерческой недвижимостью     651
жилье                                     647
покупка жилья                             647
покупка жилья для семьи                   641
строительство собственной недвижимости    635
недвижимость                              634
операции со своей недвижимостью           630
строительство жилой недвижимости          626
покупка недвижимости                      624
строительство недвижимости                620
покупка своего жилья                      620
ремонт жилью                              612
покупка жилой недвижимости                607
на покупку своего автомобиля              505
заняться высшим образованием      

**Выводы.** В данных есть ошибки, возникшие, скорее всего, вследствие действия человеческого фактора или ошибок загрузки. Столбец 'children' содержит значение (-1) 47 раз и значение (20) 76 раз, при этом нет заемщиков с количеством детей от 6 до 19. Столбец 'days_employed' содержит только отрицательные значения либо артефакты: 400281 день, 340266 дней (1361 год), даже если это рабочие часы, то это около 170 лет. Самому пожилому заемщику 75 лет, поэтому максимальное количество отработанных дней может быть около 15000: (75-18)*250. Столбец 'dob_years' также собержит 101 значение 0. Столбец 'education' содержит значения с заглавными и прописными буквами. Столбец 'gender' сдержит одно значение XNA. Столбец 'purpose' содержит много похожих целей. Все столбцы, кроме days_employed и total_income содержат 21525 ненулевых значений, а указанные два столбца содержат одинаковое количество значений, равное 19351. Скорее всего, нулевые значения в этих столбцах совпадают по строкам. Это подтверждается строками 12 и 21510. 

## Шаг 2. Предобработка данных

### Обработка пропусков

In [2]:
# Посмотрим таблицу со строками нулевых значений в столбцах 'days_employed' и 'total_income',а также их количество. 
display(data[data['days_employed'].isnull()].head())
display(data[data['days_employed'].isnull()].count())
display(data[data['total_income'].isnull()].head())
display(data[data['total_income'].isnull()].count())

children  days_employed  dob_years education  education_id  \
12         0            NaN         65   среднее             1   
26         0            NaN         41   среднее             1   
29         0            NaN         63   среднее             1   
41         0            NaN         50   среднее             1   
55         0            NaN         54   среднее             1   

            family_status  family_status_id gender  income_type  debt  \
12       гражданский брак                 1      M    пенсионер     0   
26        женат / замужем                 0      M  госслужащий     0   
29  Не женат / не замужем                 4      F    пенсионер     0   
41        женат / замужем                 0      F  госслужащий     0   
55       гражданский брак                 1      F    пенсионер     1   

    total_income                           purpose  
12           NaN                   сыграть свадьбу  
26           NaN                       образование  
29           NaN  строительство жилой недвижимости  
41           NaN  сделка с подержанным автомобилем  
55           NaN                   сыграть свадьбу

children            2174
days_employed          0
dob_years           2174
education           2174
education_id        2174
family_status       2174
family_status_id    2174
gender              2174
income_type         2174
debt                2174
total_income           0
purpose             2174
dtype: int64

children  days_employed  dob_years education  education_id  \
12         0            NaN         65   среднее             1   
26         0            NaN         41   среднее             1   
29         0            NaN         63   среднее             1   
41         0            NaN         50   среднее             1   
55         0            NaN         54   среднее             1   

            family_status  family_status_id gender  income_type  debt  \
12       гражданский брак                 1      M    пенсионер     0   
26        женат / замужем                 0      M  госслужащий     0   
29  Не женат / не замужем                 4      F    пенсионер     0   
41        женат / замужем                 0      F  госслужащий     0   
55       гражданский брак                 1      F    пенсионер     1   

    total_income                           purpose  
12           NaN                   сыграть свадьбу  
26           NaN                       образование  
29           NaN  строительство жилой недвижимости  
41           NaN  сделка с подержанным автомобилем  
55           NaN                   сыграть свадьбу

children            2174
days_employed          0
dob_years           2174
education           2174
education_id        2174
family_status       2174
family_status_id    2174
gender              2174
income_type         2174
debt                2174
total_income           0
purpose             2174
dtype: int64

In [3]:
# Посчитаем среднее арифметическое и медианы по указанным столбцам.
display(data['days_employed'].mean())
display(data['days_employed'].median())
display(data['total_income'].mean())
display(data['total_income'].median())

63046.49766147338

-1203.369528770489

167422.30220817294

145017.93753253992

Среднее арифметическое для столбца 'days_employed' составляет 56557 дней, что соответствует более 200 годам. 
При этом значения в столбце либо отрицательные, либо составляют нереальные величины. 
Для заполнения нулевых значений используем медиану.

In [4]:
# Заполним нулевые значения выбранным методом
median_days_employed = data['days_employed'].median()
data['days_employed'] = data['days_employed'].fillna(median_days_employed)
# Проверим, остались ли в таблице нулевые значения
data[data['days_employed'].isnull()].count()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

Для столбца 'total_income' значения средней арифметической и медианы отличаются незначительно, поэтому для заполнения нулевых значений используем медиану.

In [5]:
# Пройдем в цикле по уникальным значениям income_type, для каждой группы посчитаем медиану и заполним пропуски
for income_type in data['income_type'].unique():
    median = data.loc[data['income_type'] == income_type, 'total_income'].median()
    data.loc[(data['income_type'] == income_type) & (data['total_income'].isna()), 'total_income'] = median
# Проверим, остались ли в таблице нулевые значения
data[data['total_income'].isnull()].count()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

In [6]:
# Заполним пропуски в столбце total_income еще одним способом
data['total_income'] = data.groupby('income_type')['total_income'].apply(lambda x: x.fillna(x.median()))
# Проверим, остались ли в таблице нулевые значения
data[data['total_income'].isnull()].count()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

**Выводы.** Колонки 'days_employed' и 'total_income' содержат по 2174 нулевых значений. Возможными причинами появления пропусков данных могут служить нежелание клиентов сообщать сведения о своих доходах и стаже, либо их отсутствие (например, у студентов).  Удалять эти строки не стоит, т.к. это 10,1% от всех данных (2174/21525). 
Нулевые значения в ьаблице отсутствуют.

### Замена типа данных

In [7]:
# Переведем значения столбцов, содеражащих сведения о трудовом стаже заемщиков и их доходе, в целые числа
data['days_employed'] = data['days_employed'].astype('int')
data['total_income'] = data['total_income'].astype('int')
# Проверим, что получилось, распечатав первые 5 строк таблицы.
data.head()

children  days_employed  dob_years education  education_id  \
0         1          -8437         42    высшее             0   
1         1          -4024         36   среднее             1   
2         0          -5623         33   Среднее             1   
3         3          -4124         32   среднее             1   
4         0         340266         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

**Выводы.** Общий трудовой стаж измеряется в целых днях, поэтому значения в столбце 'days_employed' необходимо перевести в целые числа. Ежемесячный доход измеряется в рублях и копейках. Копейки - это 2 знака после запятой, однако даже такая точность для решения поставленной задачи не требуется, поэтому значения столбца 'total_income' также необходимо перевести в целые числа.
Дробные значения могли возникнут в таблице в следствие деления годового дохода заемщиков на 12 месяцев, а стажа, измеряемого в года и месяцах на количесво дней в году или в месяце.

### Обработка дубликатов

In [8]:
# Посчитаем количество дубликатов набором методов data.duplicated().sum().
data.duplicated().sum()

54

Результат 54. Понизим регистр в столбце 'education' и вновь посчитаем количество дубликатов.

In [9]:
data['education'] = data['education'].str.lower()
data.duplicated().sum()

71

Результат - 71. Значит, есть строки, отличающиеся только заглавными буквами в столбце 'education'. Их тоже надо удалить.

In [10]:
data = data.drop_duplicates().reset_index(drop = True)

In [11]:
# Проверим количество дубликатов и строк в таблице, посмотрим первые 5 строк таблицы.
display(data.duplicated().sum())
data.shape

0

(21454, 12)

**Выводы.** Дубликатов 0, количество строк в таблице - 21454, т.е. на 71 меньше.
Дубликаты могли возникнуть вследствие действия человеческого фактора: сотрудник дважды ввел информацию об одном клиенте, разные сотрудники ввели информацию об одном клиенте и т.д.

### Лемматизация

In [12]:
# Загрузим библиотеку для лемматизации.
from pymystem3 import Mystem
m = Mystem()

In [13]:
# сделаем список уникальных целей столбца 'purpose' и изучим его
purpose_list = data['purpose'].unique()
purpose_list

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

In [14]:
# лемматизируем каждую из целей
lemmas = []
for row in purpose_list:
    lemma = m.lemmatize(row)
    lemmas.append(lemma)
# Посмотрим и изучим полученный результат
lemmas

[['покупка', ' ', 'жилье', '\n'],
 ['приобретение', ' ', 'автомобиль', '\n'],
 ['дополнительный', ' ', 'образование', '\n'],
 ['сыграть', ' ', 'свадьба', '\n'],
 ['операция', ' ', 'с', ' ', 'жилье', '\n'],
 ['образование', '\n'],
 ['на', ' ', 'проведение', ' ', 'свадьба', '\n'],
 ['покупка', ' ', 'жилье', ' ', 'для', ' ', 'семья', '\n'],
 ['покупка', ' ', 'недвижимость', '\n'],
 ['покупка', ' ', 'коммерческий', ' ', 'недвижимость', '\n'],
 ['покупка', ' ', 'жилой', ' ', 'недвижимость', '\n'],
 ['строительство', ' ', 'собственный', ' ', 'недвижимость', '\n'],
 ['недвижимость', '\n'],
 ['строительство', ' ', 'недвижимость', '\n'],
 ['на', ' ', 'покупка', ' ', 'подержать', ' ', 'автомобиль', '\n'],
 ['на', ' ', 'покупка', ' ', 'свой', ' ', 'автомобиль', '\n'],
 ['операция', ' ', 'с', ' ', 'коммерческий', ' ', 'недвижимость', '\n'],
 ['строительство', ' ', 'жилой', ' ', 'недвижимость', '\n'],
 ['жилье', '\n'],
 ['операция', ' ', 'со', ' ', 'свой', ' ', 'недвижимость', '\n'],
 ['автомобиль'

Очевидны группы расходов на покупку автомобиля, получение образования, организацию свадьбы. Невозмодно отделить цели кредита на покупку жилья для себя или для сдачи жилья в аренду, т.к. есть цель просто "жильё". Также невозможно отделить покупку жилья от приобретения коммерческой недвижимости, т.к. есть цель "недвижимость", без уточнения ее назначения. Поэтому объединим эти цели в одну: "недвижимость".

In [15]:
# Создадим функцию по стандартизации целей получения кредита.
def purpose_standart(row):
    purpose = m.lemmatize(row['purpose'])
    if 'автомобиль' in purpose:
        return 'автомобиль'
    if 'свадьба' in purpose:
        return 'свадьба'
    if 'образование' in purpose:
        return 'образование'
    if 'недвиж' or 'жил' in purpose:
        return 'недвижимость'

# Присоедим к таблице столбец со стандартизированными целями.
data['purpose_standart'] = data.apply(purpose_standart, axis=1)        
# Посмотрим, что получилось (первые 10 строк таблицы)
data.head()

children  days_employed  dob_years education  education_id  \
0         1          -8437         42    высшее             0   
1         1          -4024         36   среднее             1   
2         0          -5623         33   среднее             1   
3         3          -4124         32   среднее             1   
4         0         340266         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose purpose_standart  
0               покупка жилья     недвижимость  
1     приобретение автомобиля       автомобиль  
2               покупка жилья     недвижимость  
3  дополнительное образование      образование  
4             сыграть свадьбу          свадьба

**Выводы.**
Сведения о заемщиках в базу данных вносили, скорее всего, разные сотрудники банка, у которых не было четких инструкций, как отражать цели получения кредита. Мы привели все цели к единообразию, сформировав 4 группы. Теперь такая информация подлежит дальнейшему анализу.

### Категоризация данных

In [16]:
# Для жальнейшего анализа разобьем заемщиков на следующие группы:
# 'нет детей' 'есть дети'
# Напишем функцию для разбиения на группы
def children_class(row):
    children = row['children']
    debt = row['debt']
    
    if children == 0:
        return 'нет детей'
                
    if children != 0:
        return 'есть дети'

# Сформируем доролнительный столбец к таблице и посмотрим ее
data['children_class'] = data.apply(children_class, axis=1)
display(data.head())

children  days_employed  dob_years education  education_id  \
0         1          -8437         42    высшее             0   
1         1          -4024         36   среднее             1   
2         0          -5623         33   среднее             1   
3         3          -4124         32   среднее             1   
4         0         340266         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose purpose_standart children_class  
0               покупка жилья     недвижимость      есть дети  
1     приобретение автомобиля       автомобиль      есть дети  
2               покупка жилья     недвижимость      нет детей  
3  дополнительное образование      образование      есть дети  
4             сыграть свадьбу          свадьба      нет детей

In [17]:
# Разделим клиентов банка по уровню дохода тыс. руб. в месяц на 3 группы: до 100 (невысокий), от 100 до 250 (средний), 
# свыше 250 (высокий) 
# Напмшем функцию для разбиения на группы
def income_class(row):
    total_income = row['total_income']
    debt = row['debt']
    if total_income < 100000:
        return 'доход невысокий'
        
    if 100000 <= total_income < 250000:
        return 'доход средний'
        
    if total_income >= 250000:
        return 'доход высокий'

# Сформируем доролнительный столбец к таблице и посмотрим ее
data['income_class'] = data.apply(income_class, axis=1)
data.head()

children  days_employed  dob_years education  education_id  \
0         1          -8437         42    высшее             0   
1         1          -4024         36   среднее             1   
2         0          -5623         33   среднее             1   
3         3          -4124         32   среднее             1   
4         0         340266         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose purpose_standart children_class   income_class  
0               покупка жилья     недвижимость      есть дети  доход высокий  
1     приобретение автомобиля       автомобиль      есть дети  доход средний  
2               покупка жилья     недвижимость      нет детей  доход средний  
3  дополнительное образование      образование      есть дети  доход высокий  
4             сыграть свадьбу          свадьба      нет детей  доход средний

**Выводы.** Мы разбили всех клиентов на группы. Полученная информация позволит нам ответить на вопросы, поставленные заказчиком.

## Шаг 3. Ответим на поставленные вопросы

- Есть ли зависимость между наличием детей и возвратом кредита в срок?

In [18]:
# Посчитаем долю заемщиков, просрочивших выплаты по кредитам, в зависимости от наличия детей
data_new = data[['children_class', 'debt']]
data_new.groupby(['children_class']).mean().style.format({'debt': '{:.2%}'})

**Выводы.** Доля просроченных кредитов у заемщиков, имеющих детей выше на 22,5%, чем у бездетных заемщиков, следоавтельно существует зависимость между наличием детей и возвратом кредита в срок.

- Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [19]:
# Посчитаем долю заемщиков, просрочивших выплаты по кредитам, в зависимости от семейного положения
data_new = data[['family_status_id', 'debt']]
data_new.groupby(['family_status_id']).mean().style.format({'debt': '{:.2%}'})

**Выводы.** Минимальная доля просроченных кредитов приходится на заемщиков - вдовцов/вдов, максимальная - на заемщиков -  Не женат / не замужем (просрочек больше на 48,4%), следовательно, существует зависимость между семейным положением и возвратом кредита в срок.

- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [20]:
# Посчитаем долю заемщиков, просрочивших выплаты по кредитам, в зависимости от уровня дохода
data_new = data[['income_class', 'debt']]
data_new.groupby(['income_class']).mean().style.format({'debt': '{:.2%}'})

**Выводы.** Минимальная доля просроченных кредитов приходится на заемщиков с высоким уровнем дохода, максимальная - на заемщиков со средним уровнем дохода (просрочек больше на 22,0%).
Вывод: существует зависимость между уровнем дохода и возвратом кредита в срок.

- Как разные цели кредита влияют на его возврат в срок?

In [21]:
# Посчитаем долю заемщиков, просрочивших выплаты по кредитам, в зависимости от цели
data_new = data[['purpose_standart', 'debt']]
data_new.groupby(['purpose_standart']).mean().style.format({'debt': '{:.2%}'})

**Выводы.** Минимальная доля просроченных кредитов приходится на цели приобретения недвижимости, максимальная - приобретения автомобилей (просрочек больше на 29,4%).
Вывод: разные цели кредита влияют на его возврат в срок.

## Шаг 4. Общий вывод

**ВЫВОД**

Идеальный заемщик: вдовец/вдова, без детей, с высоким уровнем дохода, берущий кредит на покупку недвижимости. Худший заемщик: не женатый / не замужняя, имеющий детей, со средним уровнем дохода, берущий автокредит. На стадии предобработки данных были выявлены существенные недостатки предоставленных данных, влияющие на результаты анализа, которые были устранены. Для ответа на поставленные вопросы все клиенты разбивались на группы, в результате чего выяснилось, что на возврат кредита в срок в той или иной степени влияют такие факторы, как наличие детей, семейное положение заемщиков, уровень дохода и цель получения кредита.

**РЕКОММЕНДАЦИИ** 

Банку необходимо установить бальную оценку надежности заемщиков по каждой из проанализированных групп. Также для упрощения анализа данных банку следует создать словарь видов целей кредования, а также короткую и понятную инструкцию для сотрудников по единообразному внесению информации в базу данных и недопустимости дублирования сведений.